esmvalcore/_recipe/check.py

```diff
@@ -366,20 +366,29 @@ def _check_delimiter(timerange):
 
 
 def _check_duration_periods(timerange):
-    try:
-        isodate.parse_duration(timerange[0])
-    except ValueError:
-        pass
-    else:
+    # isodate duration must always start with P
+    if timerange[0].startswith("P") and timerange[1].startswith("P"):
+        raise RecipeError(
+            "Invalid value encountered for `timerange`. "
+            "Cannot set both the beginning and the end "
+            "as duration periods."
+        )
+
+    if timerange[0].startswith("P"):
+        try:
+            isodate.parse_duration(timerange[0])
+        except isodate.isoerror.ISO8601Error as exc:
+            raise RecipeError(
+                str(exc) + "\n" + "Invalid value encountered for `timerange`. "
+                f"{timerange[0]} is not valid duration according to ISO 8601."
+            )
```

Comment:
that's a very good point - I usually do that myself, but I've noticed that the `isodate` exceptions are a bit cryptic, so I didn't want the user to be left with that as the last communication in the Traceback; if you think we sould really switch the order, then I'll do it, but otherwise I'd leave our (more friendly) exception as the last one :beer:

Reply:
At least the one from the tests seems to be clearer if you switch the order:

Now:

```
"ISO 8601 time designator 'T' missing. Unable to parse datetime string '11X'
Invalid value encountered for `timerange`. P11X is not valid duration according to ISO 8601."
```

Switched order:

```
"Invalid value encountered for `timerange`. P11X is not valid duration according to ISO 8601:
ISO 8601 time designator 'T' missing. Unable to parse datetime string '11X'"
```

So I think switching makes sense 😅

Reply:
Sounds good, buds! Lemme plug it in right after my neeting am in now 👍🍺

Reply:
OK buds done them all now - isodate exc at the end (so we have continuity) in 5670abd57c924491f2432e2b7e501be84a3e42f6 and c8f59e4c291258a4cfb096dc71bff051a08a6235